webapp/src/pages/AdminPage/AdminPage.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  AvatarPageTemplate,
+  ButtonVariant,
+  Input,
+  InputVariant,
+} from '../../shared/components';
+import { useAuth } from '../../shared/hooks/UseAuth';
+import socket from '../../shared/socket';
+import {
+  ResponseError,
+  UserToRoleDto,
+  UserToRoleDtoRoleEnum,
+} from '../../shared/generated';
+import { useCallback, useEffect, useState } from 'react';
+import { usersApi } from '../../shared/services/ApiService';
+import { WsException } from '../../shared/types';
+import { useNotificationContext } from '../../shared/context/NotificationContext';
+
+type UserToRoleDtoProps = UserToRoleDto & {
+  username: string;
+};
+export default function AdminPage() {
+  const { warn } = useNotificationContext();
+  const { authUser, isLoading } = useAuth();
+  const [userToRoleValues, setUserToRoleValues] =
+    useState<UserToRoleDtoProps | null>(null);
+  const getUserId = useCallback(
+    (username: string) => {
+      if (userToRoleValues && userToRoleValues.username !== '')
+        return usersApi.userControllerGetUserByUserName({
+          userName: username,
+        });
+      return Promise.reject(new Error('Could not get user with username'));
+    },
+    [userToRoleValues],
+  );
+  const handleOnSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    if (userToRoleValues) {
+      try {
+        const userToModify = await getUserId(userToRoleValues.username);
+        socket.emit('toggleUserWithRoles', {
+          id: userToModify.id,
+          role: UserToRoleDtoRoleEnum.Owner,
```

Comment:
To test the assignment of different roles, i'm hardcoding different values here in L44.